webapp/packages/plugin-data-import/src/DataImportService.ts

```diff
@@ -39,28 +39,51 @@ export class DataImportService {
   }
 
   async importData(connectionId: string, contextId: string, projectId: string, resultsId: string, processorId: string, file: File) {
+    let taskId: string | undefined;
+    const abortController = new AbortController();
+
+    const cancel = async () => {
+      if (taskId) {
+        await this.asyncTaskInfoService.cancel(taskId);
+      } else {
+        abortController.abort();
+      }
+    };
```

Comment:
better to have `let cancelImplementation: (() => void) | null = null`
and then set this on every step to proper cancellation function
```
let cancelImplementation: (() => void) | null = null;
function cancel() {
  cancelImplementation?.();
}
```